### ISSUE TITLE
Taint issues since last update

### ISSUE THREAD

**Issue by WanderingFox** [frozn/TipTac#80]
Ever since the last update I've been getting a set of three different taint issues. Not sure if it's tiptac or another addon, but figured I'd ask here first in case someone else has had the issue.

They seem to predominately happen when in combat and hovering over things.

GetObjectType taint
```
27x TipTac\ttCore.lua:1032: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `GetObjectType'
[string "@TipTac\ttCore.lua"]:1032: in function `ReApplyAnchorTypeForMouse'
[string "@TipTac\ttCore.lua"]:1280: in function <TipTac\ttCore.lua:1272>
[string "=[C]"]: ?
[string "=[C]"]: in function `SetShown'
[string "@Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua"]:102: in function `OnEnter'
[string "@Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua"]:590: in function <...Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua:589>

Locals:
(*temporary) = EmbeddedItemTooltip {
 0 = <userdata>
 SetQuestPartyProgress = <function> defined =[C]:-1
 ttNumLines = 0
 OnBackdropLoaded = <function> defined @SharedXML\Backdrop.lua:144
 ItemTooltip = <unnamed> {
 }
 SetHyperlink = <function> defined =[C]:-1
 SetUnit = <function> defined =[C]:-1
 SetAzeriteEssence = <function> defined =[C]:-1
 RightEdge = <unnamed> {
 }
 ttSetBackdropBorderColorLocked = true
 ClearBackdrop = <function> defined @SharedXML\Backdrop.lua:274
 SetPetAction = <function> defined =[C]:-1
 SetLFGDungeonReward = <function> defined =[C]:-1
 ttDefaultAnchored = false
 ttGradient = <unnamed> {
 }
 SetRecipeReagentItem = <function> defined =[C]:-1
 SetQuestCurrency = <function> defined =[C]:-1
 SetCurrencyTokenByID = <function> defined =[C]:-1
 SetupTextureCoordinates = <function> defined @SharedXML\Backdrop.lua:206
 BottomOverlay = <unnamed> {
 }
 ttLastUpdate = 0
 GetEdgeSize = <function> defined @SharedXML\Backdrop.lua:180
 NineSlice = <unnamed> {
 }
 SetCompanionPet = <function> defined =[C]:-1
 TopEdge = <unnamed> {
 }
 BottomRightCorner = <unnamed> {
 }
 SetBackdropColor = <function> defined @TipTac\ttCore.lua:1622
 LeftEdge = <unnamed> {
 }
 SetCurrencyByID = <function> defined =[C]:-1
 SetSpellBookItem = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 SetCurrencyToken = <function> defined =[C]:-1
 updateTooltipTimer = 0.167000
 SetBackdropBorderColor = <function> defined @TipTac\ttCore.lua:1630
 comparing = false
 ttAnchorType = "mouse"
 ttDisplayingUnit = false
 waitingForData = false
 textRight2Font = "GameTooltipText"
 ttDisplayingAura = false
 ApplyBackdrop = <function> defined @TipTac\ttCore.lua:1606
 SetBackdrop = <function> defined @TipTac\ttCore.lua:1614
 ttBackdropBorderColorApplied = false
 TopOverlay = <unnamed> {
 }
 GetBackdropBorderColor = <function> defined @SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @SharedXML\Backdrop.lua:339
 SetUnitDebuff = <function> defined =[C]:-1
 ttBackdropColorApplied = false
 TextLeft1 = EmbeddedItemTooltipTextLeft1 {
 }
 SetQuestItem = <function> defined =[C]:-1
 SetAzeriteEssenceSlot = <function> defined =[C]:-1
 SetBorderBlendMode = <function> defined @SharedXML\Backdrop.lua:258
 SetToyByItemID = <function> defined =[C]:-1
 TopLeftCorner = <unnamed> {
 }
 BottomFontString = <unnamed> {
 }
 HasBackdropInfo = <function> defined @SharedXML\Backdrop.lua:270
 BottomLeftCorner = <unnamed> {
 }
 shoppingTooltips = <table> {
 }
 GetBackdropColor = <function> defined @SharedXML\Backdrop.lua:382
 SetLFGDungeonShortageReward = <function> defined =[C]:-1
 backdropInfo = <table> {
 }
 OnBackdropSizeChanged = <function> defined @SharedXML\Backdrop.lua:174
 SetUnitBuff = <function> defined =[C]:-1
 SetEnhancedConduit = <function> defined =[C]:-1
 SetConduit = <function> defined =[C]:-1
 textRight1Font = "GameTooltipHeaderText"
 TextRight1 = EmbeddedItemTooltipTextRight1 {
 }
 ttSetBackdropLocked = true
 BottomEdge = <unnamed> {
 }
 textLeft2Font = "GameTooltipText"
 SetQuestLogItem = <function> defined =[C]:-1
 TextRight2 = EmbeddedItemTooltipTextRight2 {
 }
 TextLeft2 = EmbeddedItemTooltipTextLeft2 {
 }
 SetUnitAura = <function> defined =[C]:-1
 Center = <unnamed> {
 }
 SetAction = <function> defined =[C]:-1
 ttSetBackdropColorLocked = false
 needsReset = true
 SetupPieceVisuals = <function> defined @SharedXML\Backdrop.lua:238
 ttAnchorPoint = "BOTTOMRIGHT"
 SetQuestLogCurrency = <function> defined =[C]:-1
 TopRightCorner = <unnamed> {
 }
}
```

SetVertexColor taint
```
3x SharedXML\Backdrop.lua:397: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `SetVertexColor'
[string "@SharedXML\Backdrop.lua"]:397: in function <SharedXML\Backdrop.lua:391>
[string "@TipTac\ttCore.lua"]:1627: in function `SetBackdropColor'
[string "@TipTac\ttCore.lua"]:746: in function `SetBackdropColorLocked'
[string "@TipTac\ttCore.lua"]:774: in function `ResetBackdropBorderColorLocked'
[string "@TipTac\ttCore.lua"]:1385: in function <TipTac\ttCore.lua:1379>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: in function `Hide'
[string "@Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua"]:109: in function `OnLeave'
[string "@Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua"]:595: in function <...Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua:594>

Locals:
(*temporary) = <unnamed> {
 0 = <userdata>
}
(*temporary) = 0
(*temporary) = 0
(*temporary) = 0
(*temporary) = 0.810000
```

SetCenterColor taint
```
1x TipTac\ttCore.lua:1680: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `GetParent'
[string "@TipTac\ttCore.lua"]:1680: in function `SetCenterColor'
[string "@SharedXML\SharedTooltipTemplates.lua"]:52: in function <SharedXML\SharedTooltipTemplates.lua:41>
[string "=[C]"]: in function `SharedTooltip_SetBackdropStyle'
[string "@FrameXML\GameTooltip.lua"]:413: in function <FrameXML\GameTooltip.lua:409>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: in function `Hide'
[string "@Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua"]:109: in function `OnLeave'
[string "@Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua"]:595: in function <...Blizzard_UIWidgets\Blizzard_UIWidgetTemplateBase.lua:594>

Locals:
(*temporary) = <unnamed> {
 BottomLeftCorner = <unnamed> {
 }
 OnBackdropLoaded = <function> defined @SharedXML\Backdrop.lua:144
 GetBackdropColor = <function> defined @SharedXML\Backdrop.lua:382
 SetBackdropBorderColor = <function> defined @TipTac\ttCore.lua:1671
 RightEdge = <unnamed> {
 }
 BottomEdge = <unnamed> {
 }
 SetBackdrop = <function> defined @TipTac\ttCore.lua:1655
 GetCenterColor = <function> defined @SharedXML\NineSlice.lua:231
 GetBackdropBorderColor = <function> defined @SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @SharedXML\Backdrop.lua:339
 SetBorderBlendMode = <function> defined @SharedXML\Backdrop.lua:258
 SetCenterColor = <function> defined @TipTac\ttCore.lua:1679
 GetFrameLayoutTextureKit = <function> defined @SharedXML\NineSlice.lua:213
 SetupTextureCoordinates = <function> defined @SharedXML\Backdrop.lua:206
 GetBorderColor = <function> defined @SharedXML\NineSlice.lua:249
 TopRightCorner = <unnamed> {
 }
 0 = <userdata>
 GetEdgeSize = <function> defined @SharedXML\Backdrop.lua:180
 SetupPieceVisuals = <function> defined @SharedXML\Backdrop.lua:238
 OnLoad = <function> defined @SharedXML\NineSlice.lua:217
 TopEdge = <unnamed> {
 }
 ApplyBackdrop = <function> defined @TipTac\ttCore.lua:1647
 SetBorderColor = <function> defined @TipTac\ttCore.lua:1687
 HasBackdropInfo = <function> defined @SharedXML\Backdrop.lua:270
 OnBackdropSizeChanged = <function> defined @SharedXML\Backdrop.lua:174
 Center = <unnamed> {
 }
 TopLeftCorner = <unnamed> {
 }
 GetFrameLayoutType = <function> defined @SharedXML\NineSlice.lua:209
 BottomRightCorner = <unnamed> {
 }
 SetBackdropColor = <function> defined @TipTac\ttCore.lua:1663
 LeftEdge = <unnamed> {
 }
 ClearBackdrop = <function> defined @SharedXML\Backdrop.lua:274
}
```

**Comment by frozn** [comment]
I will have a look...

**Comment by hobulian** [comment]
Related to #76

**Comment by MichelPate** [comment]
Hi, I've something similar happening when I'm in dungeons/raid, I've the **friendly nameplates** activated which might be an issue, as their seems to be protected :

_**ttItemRef.lua**_
```lua
12x TipTacItemRef\ttItemRef.lua:2115: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `AddLine'
[string "@TipTacItemRef\ttItemRef.lua"]:2115: in function `spell'
[string "@TipTacItemRef\ttItemRef.lua"]:498: in function <TipTacItemRef\ttItemRef.lua:489>
[string "=[C]"]: in function `SetUnitAura'
[string "@Blizzard_NamePlates\Blizzard_NamePlates.lua"]:667: in function <...eBlizzard_NamePlates\Blizzard_NamePlates.lua:665>

Locals:
(*temporary) = NamePlateTooltip {
 0 = <userdata>
 SetQuestPartyProgress = <function> defined =[C]:-1
 OnBackdropLoaded = <function> defined @SharedXML\Backdrop.lua:144
 GetBackdropColor = <function> defined @SharedXML\Backdrop.lua:382
 SetHyperlink = <function> defined =[C]:-1
 SetUnit = <function> defined =[C]:-1
 SetAzeriteEssence = <function> defined =[C]:-1
 RightEdge = <unnamed> {
 }
 ttSetBackdropBorderColorLocked = false
 ClearBackdrop = <function> defined @SharedXML\Backdrop.lua:274
 SetPetAction = <function> defined =[C]:-1
 SetLFGDungeonReward = <function> defined =[C]:-1
 SetRecipeReagentItem = <function> defined =[C]:-1
 SetQuestCurrency = <function> defined =[C]:-1
 SetCurrencyTokenByID = <function> defined =[C]:-1
 SetupTextureCoordinates = <function> defined @SharedXML\Backdrop.lua:206
 TopRightCorner = <unnamed> {
 }
 GetEdgeSize = <function> defined @SharedXML\Backdrop.lua:180
 NineSlice = <unnamed> {
 }
 SetCompanionPet = <function> defined =[C]:-1
 TopEdge = <unnamed> {
 }
 BottomRightCorner = <unnamed> {
 }
 SetBackdropColor = <function> defined @TipTac\ttCore.lua:1622
 LeftEdge = <unnamed> {
 }
 SetCurrencyByID = <function> defined =[C]:-1
 SetSpellBookItem = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 SetCurrencyToken = <function> defined =[C]:-1
 updateTooltipTimer = 0.175000
 TopLeftCorner = <unnamed> {
 }
 ttAnchorType = "normal"
 TextRight1 = NamePlateTooltipTextRight1 {
 }
 ApplyBackdrop = <function> defined @TipTac\ttCore.lua:1606
 SetBackdrop = <function> defined @TipTac\ttCore.lua:1614
 TopOverlay = <unnamed> {
 }
 GetBackdropBorderColor = <function> defined @SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @SharedXML\Backdrop.lua:339
 SetUnitDebuff = <function> defined =[C]:-1
 waitingForData = false
 ttBackdropBorderColorApplied = false
 SetQuestItem = <function> defined =[C]:-1
 ttBackdropColorApplied = false
 SetBorderBlendMode = <function> defined @SharedXML\Backdrop.lua:258
 SetToyByItemID = <function> defined =[C]:-1
 ttDisplayingAura = true
 backdropInfo = <table> {
 }
 BottomLeftCorner = <unnamed> {
 }
 SetBackdropBorderColor = <function> defined @TipTac\ttCore.lua:1630
 TextLeft1 = NamePlateTooltipTextLeft1 {
 }
 HasBackdropInfo = <function> defined @SharedXML\Backdrop.lua:270
 SetLFGDungeonShortageReward = <function> defined =[C]:-1
 Center = <unnamed> {
 }
 OnBackdropSizeChanged = <function> defined @SharedXML\Backdrop.lua:174
 SetUnitBuff = <function> defined =[C]:-1
 textRight2Font = "GameTooltipText"
 SetAzeriteEssenceSlot = <function> defined =[C]:-1
 textRight1Font = "GameTooltipHeaderText"
 BottomEdge = <unnamed> {
 }
 ttSetBackdropLocked = true
 SetEnhancedConduit = <function> defined =[C]:-1
 textLeft2Font = "GameTooltipText"
 SetConduit = <function> defined =[C]:-1
 TextRight2 = NamePlateTooltipTextRight2 {
 }
 TextLeft2 = NamePlateTooltipTextLeft2 {
 }
 SetUnitAura = <function> defined =[C]:-1
 SetQuestLogItem = <function> defined =[C]:-1
 SetAction = <function> defined =[C]:-1
 ttSetBackdropColorLocked = false
 needsReset = true
 SetupPieceVisuals = <function> defined @SharedXML\Backdrop.lua:238
 ttAnchorPoint = "BOTTOMRIGHT"
 SetQuestLogCurrency = <function> defined =[C]:-1
 BottomOverlay = <unnamed> {
 }
}
(*temporary) = "Caster: Plaguebinder"
(*temporary) = 0.200000
(*temporary) = 0.600000
(*temporary) = 1
```
```lua
2x TipTacItemRef\ttItemRef.lua:293: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `GetName'
[string "@TipTacItemRef\ttItemRef.lua"]:293: in function `OnApplyConfig'
[string "@TipTacItemRef\ttItemRef.lua"]:1630: in function `?'
[string "@TipTacItemRef\ttItemRef.lua"]:199: in function <TipTacItemRef\ttItemRef.lua:199>
[string "=[C]"]: ?
[string "=[C]"]: in function `LoadAddOn'
[string "@FrameXML\UIParent.lua"]:544: in function `UIParentLoadAddOn'
[string "@FrameXML\UIParent.lua"]:710: in function `CollectionsJournal_LoadUI'
[string "@FrameXML\UIParent.lua"]:2136: in function <FrameXML\UIParent.lua:1337>
[string "=[C]"]: ?
[string "=[C]"]: in function `TurnOrActionStop'
[string "TURNORACTION"]:4: in function <[string "TURNORACTION"]:1>

Locals:
(*temporary) = NamePlateTooltip {
 0 = <userdata>
 SetQuestPartyProgress = <function> defined =[C]:-1
 OnBackdropLoaded = <function> defined @SharedXML\Backdrop.lua:144
 GetBackdropColor = <function> defined @SharedXML\Backdrop.lua:382
 SetHyperlink = <function> defined =[C]:-1
 SetUnit = <function> defined =[C]:-1
 SetAzeriteEssence = <function> defined =[C]:-1
 RightEdge = <unnamed> {
 }
 ttSetBackdropBorderColorLocked = true
 ClearBackdrop = <function> defined @SharedXML\Backdrop.lua:274
 SetPetAction = <function> defined =[C]:-1
 SetLFGDungeonReward = <function> defined =[C]:-1
 SetRecipeReagentItem = <function> defined =[C]:-1
 SetQuestCurrency = <function> defined =[C]:-1
 SetCurrencyTokenByID = <function> defined =[C]:-1
 SetupTextureCoordinates = <function> defined @SharedXML\Backdrop.lua:206
 TopRightCorner = <unnamed> {
 }
 GetEdgeSize = <function> defined @SharedXML\Backdrop.lua:180
 NineSlice = <unnamed> {
 }
 SetCompanionPet = <function> defined =[C]:-1
 TopEdge = <unnamed> {
 }
 BottomRightCorner = <unnamed> {
 }
 SetBackdropColor = <function> defined @TipTac\ttCore.lua:1622
 LeftEdge = <unnamed> {
 }
 SetCurrencyByID = <function> defined =[C]:-1
 SetSpellBookItem = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 SetCurrencyToken = <function> defined =[C]:-1
 updateTooltipTimer = 0.184000
 TopLeftCorner = <unnamed> {
 }
 ttAnchorType = "normal"
 TextRight1 = NamePlateTooltipTextRight1 {
 }
 ApplyBackdrop = <function> defined @TipTac\ttCore.lua:1606
 SetBackdrop = <function> defined @TipTac\ttCore.lua:1614
 TopOverlay = <unnamed> {
 }
 GetBackdropBorderColor = <function> defined @SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @SharedXML\Backdrop.lua:339
 SetUnitDebuff = <function> defined =[C]:-1
 waitingForData = false
 ttBackdropBorderColorApplied = false
 SetQuestItem = <function> defined =[C]:-1
 ttBackdropColorApplied = false
 SetBorderBlendMode = <function> defined @SharedXML\Backdrop.lua:258
 SetToyByItemID = <function> defined =[C]:-1
 ttDisplayingAura = true
 backdropInfo = <table> {
 }
 BottomLeftCorner = <unnamed> {
 }
 SetBackdropBorderColor = <function> defined @TipTac\ttCore.lua:1630
 TextLeft1 = NamePlateTooltipTextLeft1 {
 }
 HasBackdropInfo = <function> defined @SharedXML\Backdrop.lua:270
 SetLFGDungeonShortageReward = <function> defined =[C]:-1
 Center = <unnamed> {
 }
 OnBackdropSizeChanged = <function> defined @SharedXML\Backdrop.lua:174
 SetUnitBuff = <function> defined =[C]:-1
 textRight2Font = "GameTooltipText"
 SetAzeriteEssenceSlot = <function> defined =[C]:-1
 textRight1Font = "GameTooltipHeaderText"
 BottomEdge = <unnamed> {
 }
 ttSetBackdropLocked = true
 SetEnhancedConduit = <function> defined =[C]:-1
 textLeft2Font = "GameTooltipText"
 SetConduit = <function> defined =[C]:-1
 TextRight2 = NamePlateTooltipTextRight2 {
 }
 TextLeft2 = NamePlateTooltipTextLeft2 {
 }
 SetUnitAura = <function> defined =[C]:-1
 SetQuestLogItem = <function> defined =[C]:-1
 SetAction = <function> defined =[C]:-1
 ttSetBackdropColorLocked = false
 needsReset = true
 SetupPieceVisuals = <function> defined @SharedXML\Backdrop.lua:238
 ttAnchorPoint = "BOTTOMRIGHT"
 SetQuestLogCurrency = <function> defined =[C]:-1
 BottomOverlay = <unnamed> {
 }
}
```


_**ttCore.lua**_
```lua
43x TipTac\ttCore.lua:1680: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `GetParent'
[string "@TipTac\ttCore.lua"]:1680: in function `SetCenterColor'
[string "@SharedXML\SharedTooltipTemplates.lua"]:52: in function <SharedXML\SharedTooltipTemplates.lua:41>
[string "=[C]"]: ?
[string "=[C]"]: in function `SharedTooltip_SetBackdropStyle'
[string "@FrameXML\GameTooltip.lua"]:404: in function <FrameXML\GameTooltip.lua:400>
[string "=[C]"]: ?
[string "=[C]"]: in function `Hide'
[string "@Blizzard_NamePlates\Blizzard_NamePlates.lua"]:673: in function <...eBlizzard_NamePlates\Blizzard_NamePlates.lua:672>

Locals:
(*temporary) = <unnamed> {
 BottomLeftCorner = <unnamed> {
 }
 OnBackdropLoaded = <function> defined @SharedXML\Backdrop.lua:144
 GetBackdropColor = <function> defined @SharedXML\Backdrop.lua:382
 SetBackdropBorderColor = <function> defined @TipTac\ttCore.lua:1671
 RightEdge = <unnamed> {
 }
 BottomEdge = <unnamed> {
 }
 SetBackdrop = <function> defined @TipTac\ttCore.lua:1655
 GetCenterColor = <function> defined @SharedXML\NineSlice.lua:231
 GetBackdropBorderColor = <function> defined @SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @SharedXML\Backdrop.lua:339
 SetBorderBlendMode = <function> defined @SharedXML\Backdrop.lua:258
 SetCenterColor = <function> defined @TipTac\ttCore.lua:1679
 GetFrameLayoutTextureKit = <function> defined @SharedXML\NineSlice.lua:213
 SetupTextureCoordinates = <function> defined @SharedXML\Backdrop.lua:206
 GetBorderColor = <function> defined @SharedXML\NineSlice.lua:249
 TopRightCorner = <unnamed> {
 }
 0 = <userdata>
 GetEdgeSize = <function> defined @SharedXML\Backdrop.lua:180
 SetupPieceVisuals = <function> defined @SharedXML\Backdrop.lua:238
 OnLoad = <function> defined @SharedXML\NineSlice.lua:217
 TopEdge = <unnamed> {
 }
 ApplyBackdrop = <function> defined @TipTac\ttCore.lua:1647
 SetBorderColor = <function> defined @TipTac\ttCore.lua:1687
 HasBackdropInfo = <function> defined @SharedXML\Backdrop.lua:270
 OnBackdropSizeChanged = <function> defined @SharedXML\Backdrop.lua:174
 Center = <unnamed> {
 }
 TopLeftCorner = <unnamed> {
 }
 GetFrameLayoutType = <function> defined @SharedXML\NineSlice.lua:209
 BottomRightCorner = <unnamed> {
 }
 SetBackdropColor = <function> defined @TipTac\ttCore.lua:1663
 LeftEdge = <unnamed> {
 }
 ClearBackdrop = <function> defined @SharedXML\Backdrop.lua:274
}
```

```lua
11x TipTac\ttCore.lua:632: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `GetName'
[string "@TipTac\ttCore.lua"]:632: in function <TipTac\ttCore.lua:631>
[string "@TipTac\ttCore.lua"]:693: in function `ApplySettings'
[string "@TipTacOptions\ttOptions.lua"]:596: in function `SetConfigValue'
[string "@TipTacOptions\libs\AzOptionsFactory.lua"]:610: in function <...rfaceTipTacOptions\libs\AzOptionsFactory.lua:609>

Locals:
(*temporary) = NamePlateTooltip {
 0 = <userdata>
 SetQuestPartyProgress = <function> defined =[C]:-1
 OnBackdropLoaded = <function> defined @SharedXML\Backdrop.lua:144
 GetBackdropColor = <function> defined @SharedXML\Backdrop.lua:382
 SetHyperlink = <function> defined =[C]:-1
 SetUnit = <function> defined =[C]:-1
 SetAzeriteEssence = <function> defined =[C]:-1
 RightEdge = <unnamed> {
 }
 ttSetBackdropBorderColorLocked = false
 ClearBackdrop = <function> defined @SharedXML\Backdrop.lua:274
 SetPetAction = <function> defined =[C]:-1
 SetLFGDungeonReward = <function> defined =[C]:-1
 SetRecipeReagentItem = <function> defined =[C]:-1
 SetQuestCurrency = <function> defined =[C]:-1
 SetCurrencyTokenByID = <function> defined =[C]:-1
 SetupTextureCoordinates = <function> defined @SharedXML\Backdrop.lua:206
 TopRightCorner = <unnamed> {
 }
 GetEdgeSize = <function> defined @SharedXML\Backdrop.lua:180
 NineSlice = <unnamed> {
 }
 SetCompanionPet = <function> defined =[C]:-1
 TopEdge = <unnamed> {
 }
 BottomRightCorner = <unnamed> {
 }
 SetBackdropColor = <function> defined @TipTac\ttCore.lua:1622
 LeftEdge = <unnamed> {
 }
 SetCurrencyByID = <function> defined =[C]:-1
 SetSpellBookItem = <function> defined =[C]:-1
 textLeft1Font = "GameTooltipHeaderText"
 SetCurrencyToken = <function> defined =[C]:-1
 updateTooltipTimer = 0.083000
 TopLeftCorner = <unnamed> {
 }
 ttAnchorType = "normal"
 TextRight1 = NamePlateTooltipTextRight1 {
 }
 ApplyBackdrop = <function> defined @TipTac\ttCore.lua:1606
 SetBackdrop = <function> defined @TipTac\ttCore.lua:1614
 TopOverlay = <unnamed> {
 }
 GetBackdropBorderColor = <function> defined @SharedXML\Backdrop.lua:401
 GetBackdrop = <function> defined @SharedXML\Backdrop.lua:339
 SetUnitDebuff = <function> defined =[C]:-1
 waitingForData = false
 ttBackdropBorderColorApplied = false
 SetQuestItem = <function> defined =[C]:-1
 ttBackdropColorApplied = false
 SetBorderBlendMode = <function> defined @SharedXML\Backdrop.lua:258
 SetToyByItemID = <function> defined =[C]:-1
 ttDisplayingAura = true
 backdropInfo = <table> {
 }
 BottomLeftCorner = <unnamed> {
 }
 SetBackdropBorderColor = <function> defined @TipTac\ttCore.lua:1630
 TextLeft1 = NamePlateTooltipTextLeft1 {
 }
 HasBackdropInfo = <function> defined @SharedXML\Backdrop.lua:270
 SetLFGDungeonShortageReward = <function> defined =[C]:-1
 Center = <unnamed> {
 }
 OnBackdropSizeChanged = <function> defined @SharedXML\Backdrop.lua:174
 SetUnitBuff = <function> defined =[C]:-1
 textRight2Font = "GameTooltipText"
 SetAzeriteEssenceSlot = <function> defined =[C]:-1
 textRight1Font = "GameTooltipHeaderText"
 BottomEdge = <unnamed> {
 }
 ttSetBackdropLocked = true
 SetEnhancedConduit = <function> defined =[C]:-1
 textLeft2Font = "GameTooltipText"
 SetConduit = <function> defined =[C]:-1
 TextRight2 = NamePlateTooltipTextRight2 {
 }
 TextLeft2 = NamePlateTooltipTextLeft2 {
 }
 SetUnitAura = <function> defined =[C]:-1
 SetQuestLogItem = <function> defined =[C]:-1
 SetAction = <function> defined =[C]:-1
 ttSetBackdropColorLocked = false
 needsReset = true
 SetupPieceVisuals = <function> defined @SharedXML\Backdrop.lua:238
 ttAnchorPoint = "BOTTOMRIGHT"
 SetQuestLogCurrency = <function> defined =[C]:-1
 BottomOverlay = <unnamed> {
 }
}
```
_**SharedXML\Backdrop.lua**_
```lua
43x SharedXML\Backdrop.lua:422: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `SetVertexColor'
[string "@SharedXML\Backdrop.lua"]:422: in function <SharedXML\Backdrop.lua:414>
[string "@TipTac\ttCore.lua"]:1635: in function `SetBackdropBorderColor'
[string "@TipTacItemRef\ttItemRef.lua"]:1869: in function `SetBackdropBorderColorLocked'
[string "@TipTacItemRef\ttItemRef.lua"]:2158: in function `spell'
[string "@TipTacItemRef\ttItemRef.lua"]:498: in function <TipTacItemRef\ttItemRef.lua:489>
[string "=[C]"]: in function `SetUnitAura'
[string "@Blizzard_NamePlates\Blizzard_NamePlates.lua"]:667: in function <...eBlizzard_NamePlates\Blizzard_NamePlates.lua:665>

Locals:
(*temporary) = <unnamed> {
 0 = <userdata>
}
(*temporary) = 0.443137
(*temporary) = 0.835294
(*temporary) = 1
(*temporary) = 1

-----

62x SharedXML\Backdrop.lua:397: Attempt to access forbidden object from code tainted by an AddOn
[string "=[C]"]: in function `SetVertexColor'
[string "@SharedXML\Backdrop.lua"]:397: in function <SharedXML\Backdrop.lua:391>
[string "@TipTac\ttCore.lua"]:1627: in function `SetBackdropColor'
[string "@TipTac\ttCore.lua"]:746: in function `SetBackdropColorLocked'
[string "@TipTac\ttCore.lua"]:774: in function `ResetBackdropBorderColorLocked'
[string "@TipTac\ttCore.lua"]:1385: in function <TipTac\ttCore.lua:1379>
[string "=[C]"]: ?
[string "=[C]"]: ?

Locals:
(*temporary) = <unnamed> {
 0 = <userdata>
}
(*temporary) = 0.078431
(*temporary) = 0.078431
(*temporary) = 0.078431
```

**Comment by frozn** [comment]
Should be fixed with latest release [v23.01.23](https://github.com/frozn/TipTac/releases/tag/v23.01.23-release). Please try again with this version.
